src/screens/IndexScreen.js

```diff
@@ -177,13 +183,20 @@ export const IndexScreen = ({ navigation }) => {
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
   const query = navigation.getParam('query', '');
   const [queryVariables, setQueryVariables] = useState(navigation.getParam('queryVariables', {}));
+  const [refreshing, setRefreshing] = useState(false);
 
   if (!query) return null;
 
   useEffect(() => {
     isConnected && auth();
   }, []);
 
+  const refresh = async (refetch) => {
+    setRefreshing(true);
+    await refetch();
+    setRefreshing(false);
+  };
+
```

Comment:
refresh is async because refetch from Apollo is async and we need to await it in order to go on. we do not wait before refetch, we wait for the refetch to be done.

in the documentation from React Native the 2 seconds are chosen randomly to simulate an async call that happens (and taking two seconds) before saying refreshing is done with setting refreshing to false.